api/v1alpha1/utils/commons.go

```diff
@@ -153,18 +183,30 @@ func (kc *KubeAuthConfiguration) GetServiceAccountName() string {
 	return kc.ServiceAccount.Name
 }
 
+func (kc *KubeAuthConfiguration) getCacheKey() string {
+	return fmt.Sprintf("%s:%s:%s:%s", kc.ServiceAccount.Name, kc.Path, kc.Role, kc.Namespace)
```

Comment:
the kube namespace needs to be in key otherwise you could have two authentication config in different namespace obtaining the same vault account

Reply:
I thought `kc.Namespace` was the kube namespace, is it not?

Reply:
no, that is the vault namespace.